design/milestone_1_2_design_doc.md

```diff
@@ -0,0 +1,455 @@
+
+|      | Solution                                                     | Pros                                                         | Cons                                                         | Effort Estimation |
+| ---- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | ----------------- |
+| 1    | Keep using only **namespace** and/or **service account** granularities. | \- Since Secrets Provider runs as a separate entity, we suggest creating a dedicated service account for Secrets Provider.<br />That way, the **service account** granularity serves as an exact match and **Job** granularity is redundant<br />- No code changes | \- Service account might be reused, which allows other apps to authenticate using Secrets Provider host<br />- Break application identity convention | Free              |
+| 2    | Add support for **Job** granularity in Conjur                | Allows specific identification of the Secrets Provider **Job**, which enhances security | \- Costly; requires code changes, verify compatibility, tests, docs<br />- Redundant if **service account** serves only the Secrets Provider | 10 days           |
+
+*Decision*: Solution #1, use **namespace** and **service account** to define the Secrets Provider host in Conjur policy. We would recommend to the customer to create a dedicated service account for Secrets Provider.
+
+#### Customer experience
+
+Deployment for the Secrets Provider will be done using Helm. That way, if the 
+
+1. Configure K8s Secrets with ***conjur-map\*** metadata defined
+
+2. Create `values.yaml` that will contain the following parameters and their defaults:
+
+   For a full list of required parameters, see `values.yaml` below
+
+   *When parameters are not configured by customer, their defaults are used*
+
+3. Install Helm Chart for the Secrets Provider
+
+We will supply the customer with a  `values.yaml` file where all default parameters are collected. Our `values.yaml` will resemble the following:
+
+```
+# K8s permissioning defaults
+permission:
+	serviceAccountName: secrets-provider-account
+	roleName: secrets-provider-role
+	roleBinding: secrets-provider-role-binding
+
+# Secrets Provider pod defaults
+secrets-provider:
+	image: cyberark/secrets-provider-for-k8s
+	name: cyberark-secrets-provider
+	namespace: default
+	ttlSecondsAfterFinished: 0	# Length at which Job should stay alive even after finishing task. Used for logging.
+	
+# Secrets Provider environment variable defaults
+secrets-provider-environment:
+	conjurApplianceUrl: https://conjur-follower.default.svc.cluster.local
+	conjurAuthnUrl: https://conjur-follower.default.svc.cluster.local/authn-k8s/authn-default
+	conjurAuthnLogin: host/conjur/authn-k8s/authn-default/default
+	conjurAccount: default
+	conjurSslCertificate: conjur-master-ca-env
+	debug=true
```

Comment:
is it a good idea to have debug logs enabled by default?

Reply:
This is a tricky one because any change to the customer's `custom-values.yaml` demands a restart of the HELM Chart which means tearing down the Secrets Provider...
So if they have a problem in their env and they want to see debug logs, they cannot unless they do so. I guess that this aligns with the current experience for the secrets provider no?

Seems like a wonky experience for this one which is why I wanted to default to debug.. What do you think?

Reply:
debug default value should be false. Regarding thaging debug level on the fly - it should be possible with some changes i n secret provider if we provide it as a file and not as env var

Reply:
We need to challenge this. I think the default need to be false.

Reply:
i also agree that the default should be false. we can find a solution to reload the debug level on runtime if we want (do you have an idea?)

Reply:
default can be false but if they supply us with `debug: true` when loading the HELM, we will update. Adding the following:
```
{{- if eq .Values.secrets-provider-environment.debug "true" }}
   - name: DEBUG
      value: true
{{- end }} 
```